### ISSUE TITLE
Multi-platform: Use tempdir() from Golang instead of /tmp

### ISSUE THREAD

**Issue by alexellis** [openfaas/faas#245]
This is a change for the watchdog.

We write a lock-file to `/tmp/` and have to disable it for Windows (via `suppress_lock`) since it is an invalid path. If we use Golang's tempdir() method in `os` then we should never need to use the `suppress_lock` configuration.

**Comment by benjic** [comment]
Would a contributor need access to a windows machine in order to test the changes?

**Comment by alexellis** [comment]
I.e. An azure VM or Vagrant box would do fine too.

**Comment by alexellis** [comment]
Change would go here:

https://github.com/openfaas/faas/blob/master/watchdog/main.go#L258

TempDir function:

https://godoc.org/os#TempDir

Suggestion by @jstarks

**Comment by fharding1** [comment]
@alexellis I believe this issue can be closed Since PR #246 was merged.

**Comment by alexellis** [comment]
It is merged and released in 0.4.17.. let's download it and give it a shot on Windows 👍

**Comment by alexellis** [comment]
MacOS:

```
fprocess=cat ./watchdog 
2017/10/01 22:30:15 Writing lock-file to: /var/folders/2f/24z_m1fj6ns4tq6pxrnb7k140000gp/T/.lock
```

Linux container with (Alpine):

```
# fprocess=cat ./watchdog 
2017/10/01 21:31:13 Writing lock-file to: /tmp/.lock
```
